src/auth/ha-password-manager-polyfill.ts

```diff
@@ -0,0 +1,90 @@
+import "@polymer/iron-input";
+import { html, LitElement, TemplateResult } from "lit";
+import { customElement, property } from "lit/decorators";
+import { fireEvent } from "../common/dom/fire_event";
+import { HaFormSchema } from "../components/ha-form/ha-form";
+import { DataEntryFlowStep } from "../data/data_entry_flow";
+
+const ENABLED_HANDLERS = [
+  "homeassistant",
+  "legacy_api_password",
+  "command_line",
+];
+
+@customElement("ha-password-manager-polyfill")
+export class HaPasswordManagerPolyfill extends LitElement {
+  @property() public step?: DataEntryFlowStep;
+
+  @property() public stepData: any;
+
+  protected createRenderRoot() {
+    // Add under document body so the element isn't placed inside any shadow roots
+    return document.body;
+  }
+
+  // Making this static for Lit doesn't work since Lit places these in the shadow dom
+  styles = `
+    .password-manager-polyfill {
+      position: absolute;
+      top: 170px;
+      left: 50%;
+      width: 0;
+      height: 0;
+      overflow: hidden;
+    }
+    .password-manager-polyfill input {
+      width: 210px;
+      height: 60px;
+    }
+  `;
+
+  protected render(): TemplateResult {
+    if (
+      this.step &&
+      this.step.type === "form" &&
+      this.step.step_id === "init" &&
+      ENABLED_HANDLERS.includes(this.step.handler[0])
+    ) {
+      return html`
+        <form
+          class="password-manager-polyfill"
+          aria-hidden="true"
+          @submit=${this._handleSubmit}
+        >
+          ${this.step.data_schema.map((input) => this.render_input(input))}
+          <input type="submit" />
+          <style>
+            ${this.styles}
+          </style>
+        </form>
+      `;
+    }
+    return html``;
+  }
+
+  private render_input(schema: HaFormSchema): TemplateResult {
+    const inputType = schema.name.includes("password") ? "password" : "text";
+    // Iron-input needed for reliable updating of the input value
+    if (schema.type === "string") {
+      return html`<iron-input bind-value=${this.stepData[schema.name]}>
```

Comment:
Why exactly do you need `iron-input`?

Reply:
My bad. The input element wasn't working because I supplied `value` instead of `.value`. I've changed it to `.value` so `iron-input` isn't needed anymore.